lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -113,78 +117,82 @@ def serverset_decode(data)
     end
 
     # helper method that ensures that the discovery path exists
-    def create(path)
+    def create(path, recursion = false)
       log.debug "synapse: creating ZK path: #{path}"
 
-      # recurse if the parent node does not exist
-      create File.dirname(path) unless @zk.exists? File.dirname(path)
-      @zk.create(path, ignore: :node_exists)
+      statsd.time('synapse.zk_watcher.create.time') do
+        # recurse if the parent node does not exist
+        create(File.dirname(path), true) unless @zk.exists? File.dirname(path)
```

Comment:
I agree. The extra parameter was intended to avoid timing at all steps throughout the recursion, but I forgot to actually act on in. It also complicates the code.

I considered renaming current `create` into an implementation method and creating an entry point that would time that, but I decided to simply time it at the call site. It makes everything simpler and renders this PR less intrusive.